### ISSUE TITLE
remove `pytorch-lightning` dependency

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pie-core#52]
we just use `HyperparametersMixin`, but `pytorch-lightning` is quite heavy. We should just re-implement `HyperparametersMixin` on our own. 

EDIT: We also use `Metric` from `torchmetrics` (but just for an interface definition).

Implementation:
- [x] #57
- [ ] #56
